frontend/src/bundles/password-reset/pages/password-reset.tsx

```diff
@@ -0,0 +1,97 @@
+import authLogo from '~/assets/img/auth-logo.svg';
+import { ResetPasswordForm } from '~/bundles/common/components/components.js';
+import { AppRoute, DataStatus } from '~/bundles/common/enums/enums.js';
+import {
+    getUserId,
+    getValidClassNames,
+} from '~/bundles/common/helpers/helpers.js';
+import {
+    useAppDispatch,
+    useAppSelector,
+    useCallback,
+    useEffect,
+    useNavigate,
+    useParams,
+    useState,
+} from '~/bundles/common/hooks/hooks.js';
+import { actions as passwordResetActions } from '~/bundles/password-reset/store/password-reset.js';
+import {
+    type PasswordResetPayload,
+    type PasswordResetRequestDto,
+} from '~/bundles/password-reset/types/types.js';
+
+import { PasswordResetSuccessMessage } from '../components/components.js';
+
+const PasswordReset: React.FC = () => {
+    const dispatch = useAppDispatch();
+    const navigate = useNavigate();
+    const { resetToken } = useParams();
+    const { userId } = getUserId(resetToken as string);
+
+    const [isPasswordReset, setIsPasswordReset] = useState(false);
+
+    const { dataStatus } = useAppSelector(({ passwordReset }) => ({
+        dataStatus: passwordReset.dataStatus,
+    }));
+
+    const isLoading = dataStatus === DataStatus.PENDING;
+
+    const handleResetPassword = useCallback(
+        (payload: PasswordResetPayload): void => {
+            if (!userId) {
+                return;
+            }
+
+            const resetPayload = {
+                id: Number(userId),
+                token: resetToken,
+                ...payload,
+            };
+            void dispatch(
+                passwordResetActions.resetPassword(
+                    resetPayload as PasswordResetRequestDto,
+                ),
+            );
+        },
+        [dispatch, resetToken, userId],
+    );
+
+    const handleCancel = useCallback((): void => {
+        void navigate(AppRoute.SIGN_IN);
+    }, [navigate]);
+
+    useEffect(() => {
+        if (dataStatus === DataStatus.FULFILLED) {
+            setIsPasswordReset(true);
+        }
+    }, [dataStatus, navigate]);
+
+    const classes = {
+        base: 'relative flex flex-col flex-1 bg-lm-black-200 mx-[1rem] my-[1.125rem] rounded-[2.75rem] lg:flex-none lg:w-[44rem]',
+        form: 'gap-6 text-white px-[2rem] pb-[3.75rem] pt-[10rem] lg:px-[11.25rem]',
+    };
+
+    return (
+        <main className="bg-auth flex h-screen flex-col-reverse bg-cover bg-no-repeat lg:flex-row">
```

Comment:
It is duplicated code from auth.tsx. I don't think we need separate bundle for password reset as it is part of auth bundle. I suggest moving this component in **auth/components** and returning just elements of form like sign-in or sign-up form so you will not duplicated styles

Reply:
In our case, we have auth page with 3 different screens - sign in form, sign up form and password reset. I think it will be more clean than having another bundle with duplicated layout

Reply:
In auth we use switch/case and the URL is "/reset-password/:token", so, in this case, it trying to find this value but we have dynamically formed link and the component does not render

Reply:
And duplication is just because we decided to have this page with the same design

Reply:
> In auth we use switch/case and the URL is "/reset-password/:token", so, in this case, it trying to find this value but we have dynamically formed link and the component does not render

It possible to use route with query parameter like /password-reset?token=TOKEN and use **pathname** and **search** in useLocation to get pathname and parameters separately
![image](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/26843161/76673264-a4d1-481b-887e-b3bc435696a8)